contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -255,13 +264,9 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
         virtual
         returns (uint256 bptExpected)
     {
-        address priceProvider = IVault(vaultAddress).priceProvider();
-        uint256 strategyAssetMarketPrice = IOracle(priceProvider).price(_asset);
         uint256 bptRate = IRateProvider(platformAddress).getRate();
-
-        bptExpected = _amount
-            .mulTruncate(strategyAssetMarketPrice)
-            .divPrecisely(bptRate);
+        uint256 poolAssetRate = getRateProviderRate(_asset);
+        bptExpected = _amount.mulTruncate(poolAssetRate).divPrecisely(bptRate);
```

Comment:
We don't have to change anything here but,

This: 

```solidity
bptExpected = _amount.mulTruncate(poolAssetRate).divPrecisely(bptRate);
```

is doing this:

```solidity
bptExpected = ((_amount * poolAssetRate / 1e18) * 1e18  / bptRate);
```

Which is the same as

```solidity
bptExpected = _amount * poolAssetRate / bptRate
```

This stable math library we are using is also using the safe math library under the hood, which adds unneeded overhead to the contracts.

Again, no change needed, just something to think about for the future.

Reply:
I haven't though of that, good point. The mulTruncate +  divPrecisely could cancel the 1e18 multiplication & division out.

We could do that change, though adding a comment would probably make sense, since the though process can easily be forgotten.